discovery_engine_core/providers/src/query.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#![allow(clippy::module_name_repetitions)]
+
+use chrono::Utc;
+
+use crate::{Filter, Market};
+
+/// Elements shared between various headlines and news queries.
+///
+/// Fields not supported by the used provider will be ignored.
+pub struct CommonQueryParts<'a> {
+    /// Market of news.
+    pub market: Option<&'a Market>,
+
+    /// How many articles to return (per page).
+    pub page_size: usize,
+
+    /// The number of the page which should be returned.
+    ///
+    /// Paging starts with `1`.
+    pub page: usize,
+
+    /// Exclude given sources.
+    pub excluded_sources: &'a [String],
+
+    /// Prefer trusted sources
+    ///
+    /// Supported by
+    /// - newscatcher/search
+    /// - newscatcher/headlines
+    pub trusted_sources: &'a [String],
+
+    /// News filter.
+    ///
+    /// Supported by
+    /// - newscatcher/search
+    /// - gnews/search
+    /// - gnews/headlines
+    pub filter: Option<&'a Filter>,
+
+    /// Headlines topic.
+    ///
+    /// Supported by
+    /// - newscatcher/search
+    /// - newscatcher/headlines
+    /// - gnews/headlines
+    pub topic: Option<&'a str>,
+}
+
+/// Parameters determining which news to fetch
+///
+/// Fields not supported by the used provider will be ignored.
+pub struct NewsQuery<'a> {
+    /// Common parts
+    pub common: CommonQueryParts<'a>,
+
+    //FIXME gnews support for from
+    /// Starting point in time from which to start the search.
+    /// The format is YYYY/mm/dd. Default timezone is UTC.
+    /// Defaults to the last week.
```

Comment:
I don't  know it's not code written by me, it would have been moved in the previous PR but  I had  overlooked it.

If I had to guess it's either outdated or the default when not given is 7days.

But in TY-2859 I will have to touch/change it anyway so I do not plan to touch it in this PR hence the FIXME :wink:.